boards/PDM/Inc/Io/Io_Efuse.h

```diff
@@ -0,0 +1,725 @@
+ * disabled CSNS_FUNCTION_CURRENT_CH0 - Current sensing for channel 0
+ *                  CSNS_FUNCTION_CURRENT_CH1 - Current sensing for channel 1
+ *                  CSNS_FUNCTION_TEMPERATURE - Ffuse temperature sensing
+ *                  CSNS_FUNCTION_CURRENT_SUM - Current sensing for summed
+ * channels (for Parallel mode)
+ * @param e_fuse Pointer to Efuse structure being configured
+ */
+void Io_Efuse_ConfigureChannelMonitoring(
+    uint8_t       selection,
+    struct Efuse *e_fuse);
+
+/**
+ * Gets the value of the Efuse's Status Register over SPI and updates the Efuse
+ * handle's operating status
+ * @param e_fuse Pointer to Efuse structure being checked
+ */
+void Io_Efuse_UpdateStatus(struct Efuse *e_fuse);
+
+/**
+ * Get the value of the Efuse's Fault Register over SPI and updates the Efuse
+ * fault status
+ * @param e_fuse Pointer to Efuse structure being checked
+ */
+void Io_Efuse_UpdateFaults(struct Efuse *e_fuse);
+
+/**
+ * Write data to a specific Serial Input register on the Efuse
+ * @param register_address Serial Input register being written to
+ * @param register_value Value being written to the Serial Input register
+ * @param e_fuse Pointer to Efuse structure being written to
+ */
+void Io_Efuse_WriteReg(
+    uint8_t       register_address,
+    uint16_t      register_value,
+    struct Efuse *e_fuse);
+
+/**
+ * Read data from a specific Serial Output register on the Efuse
+ * @param register_address Serial Output register being read from
+ * @param e_fuse Pointer to Efuse structure being read from
+ * @return Contents of the addressed register
+ */
+uint16_t Io_Efuse_ReadReg(uint8_t register_address, struct Efuse *e_fuse);
+
+/**
+ * Calculates the parity of the SPI command to be sent to the Efuse and
+ * sets/clears the parity bit. Using the XOR sum of bits method taken from:
+ * https://en.wikipedia.org/wiki/Parity_bit#Parity
+ * @param spi_command Original SPI command without the parity bit
+ */
+void Io_Efuse_SetParityBit(uint16_t *spi_command);
+
+/**
+ * Write SPI data to Efuse in blocking mode
+ * @param tx_data Pointer to data being sent to the Efuse
+ * @param chip_select_port Handle to Efuse's chip-select GPIO port
+ * @param chip_select_pin Efuse's chip-select GPIO pin
+ * @return Status of the SPI transfer
+ */
+HAL_StatusTypeDef Io_Efuse_WriteToEfuse(
+    uint16_t *    tx_data,
+    GPIO_TypeDef *chip_select_port,
+    uint16_t      chip_select_pin);
+
+/**
+ * Read SPI data from Efuse in blocking mode
+ * @param tx_data Pointer to data being sent to the Efuse
+ * @param rx_data Pointer to data being received from the Efuse
+ * @param chip_select_port Handle to Efuse's chip-select GPIO port
+ * @param chip_select_pin Efuse's chip-select GPIO pin
+ * @return Status of the SPI transfer
+ */
+HAL_StatusTypeDef Io_Efuse_ReadFromEfuse(
+    uint16_t *    tx_data,
+    uint16_t *    rx_data,
+    GPIO_TypeDef *chip_select_port,
+    uint16_t      chip_select_pin);
```

Comment:
With this change, we should just be passing in a efuse struct rather than the pin and port
```suggestion
HAL_StatusTypeDef Io_Efuse_ReadFromEfuse(
   struct Efuse* efuse,
    uint16_t *    tx_data,
    uint16_t *    rx_data);
```

Reply:
Io_Efuse_ReadFromEfuse() should be static and shouldn't be used directly, it's just a wrapper function for using SPI. I don't think we need to pass the whole struct to it?